src/ngx_http_lua_headers_in.c

```diff
@@ -658,6 +658,10 @@ ngx_http_lua_set_input_header(ngx_http_request_t *r, ngx_str_t key,
 
     dd("set header value: %.*s", (int) value.len, value.data);
 
+    key.len = ngx_http_lua_safe_header_value_len(key.data, key.len);
+
+    value.len = ngx_http_lua_safe_header_value_len(value.data, value.len);
```

Comment:
Let's check if the return value is `(size_t) -1`. If yes, then return `NGX_ERROR` in this function.